lib/block_view_dao_coin_limit_order.go

```diff
@@ -1305,11 +1305,9 @@ func _calculateDAOCoinsTransferredInLimitOrderMatch(
 	if matchingOrder.OperationType == DAOCoinLimitOrderOperationTypeASK {
 		// The matching order's quantity represents their selling coin.
 		// Which is equivalent to the transactor's buying coin.
-		transactorBuyingCoinBaseUnitsTransferred := matchingOrder.QuantityToFillInBaseUnits
+		transactorBuyingCoinBaseUnitsTransferred := matchingOrder.QuantityToFillInBaseUnits.Clone()
 
-		transactorSellingCoinBaseUnitsTransferred, err := ComputeBaseUnitsToBuyUint256(
-			matchingOrder.ScaledExchangeRateCoinsToSellPerCoinToBuy,
-			matchingOrder.QuantityToFillInBaseUnits)
+		transactorSellingCoinBaseUnitsTransferred, err := matchingOrder.BaseUnitsToBuyUint256()
```

Comment:
confirming that `matchingOrder.BaseUnitsToBuyUint256()` is equivalent to `ComputeBaseUnitsToBuyUint256(matchingOrder.ScaledExchangeRateCoinsToSellPerCoinToBuy, matchingOrder.QuantityToFillInBaseUnits)`